File: MazePicSolver/Helpers/FileIoHelper.vb
Imports System.IO

Namespace Helpers
    Public Module FileIoHelper
        Private _pathSeparator As Char = Nothing
        Private ReadOnly Property PathSeparatorCharacter() As Char
            Get
                If _pathSeparator <> Nothing Then
                    Return _pathSeparator
                End If
                _pathSeparator = Path.Combine(" ", " ").Trim().First()
                Return _pathSeparator
            End Get
        End Property

        Public Function ExtractDirectory(path As String) As String
            Return String.Join(PathSeparatorCharacter, path.Split(PathSeparatorCharacter).Take(path.Split(PathSeparatorCharacter).Length - 1))
        End Function
        Public Function ExtractFileName(path As String) As String
            Return path.Split(PathSeparatorCharacter).Last()
        End Function
    End Module
End Namespace

File: MazePicSolverCSharp/Helpers/FileIoHelper.cs
using System.IO;
using System.Linq;

namespace MazePicSolverCSharp
{
    public class FileIoHelper {
        
        private static char? _pathSeparator = null;
        
        private static char PathSeparatorCharacter {
            get {
                if ((_pathSeparator != null)) {
                    return _pathSeparator.Value;
                }
                
                _pathSeparator = Path.Combine(" ", " ").Trim().First();
                return _pathSeparator.Value;
            }
        }
        
        public static string ExtractDirectory(string path) {
            return string.Join(PathSeparatorCharacter.ToString(), path.Split(PathSeparatorCharacter).Take((path.Split(PathSeparatorCharacter).Length - 1)));
        }
        
        public static string ExtractFileName(string path) {
            return path.Split(PathSeparatorCharacter).Last();
        }
    }
}